cluster/state.go

```diff
@@ -128,7 +149,19 @@ func saveStateToKubernetes(ctx context.Context, kubeClient *kubernetes.Clientset
 	}
 }
 
-func getStateFromKubernetes(ctx context.Context, kubeClient *kubernetes.Clientset, kubeConfigPath string) *Cluster {
+func saveStateToNodes(ctx context.Context, uniqueHosts []*hosts.Host, clusterState *v3.RancherKubernetesEngineConfig, alpineImage string, prsMap map[string]v3.PrivateRegistry) error {
+	log.Infof(ctx, "[state] Saving cluster state to cluster nodes")
+	clusterFile, err := yaml.Marshal(*clusterState)
+	if err != nil {
+		return err
+	}
+	for _, host := range uniqueHosts {
+		pki.DeployStateOnPlaneHost(ctx, host, alpineImage, prsMap, string(clusterFile))
```

Comment:
doesn't look like we check for error here? should we? given it's in the method signature